sky/skylet/providers/lambda_cloud/node_provider.py

```diff
@@ -177,11 +198,31 @@ def node_tags(self, node_id: str) -> Dict[str, str]:
 
     def external_ip(self, node_id: str) -> str:
         """Returns the external ip of the given node."""
-        return self._get_cached_node(node_id=node_id)['external_ip']
+        ip = self._get_cached_node(node_id=node_id)['external_ip']
+        with ux_utils.print_exception_no_traceback():
+            if ip is None:
+                raise lambda_utils.LambdaCloudError(
+                    'A node ip address was not found. Either '
+                    '(1) Lambda Cloud has internally errored, or '
+                    '(2) the cluster is still booting. '
+                    'You can manually terminate the cluster on the '
+                    'Lambda Cloud console or (in case 2) wait for '
+                    'booting to finish (~2 minutes).')
```

Comment:
Why would this triggerred? It seems the `_get_filtered_nodes` has checked for the `ip` already?

Reply:
I've made `_get_filtered_nodes` set `ip` to `None` if ip is not found. This way, `_get_filtered_nodes` won't throw an error if `ip` is not found, which allows us to `sky down` even if `ip` doesn't exist.